deepseg.cc

```diff
@@ -257,7 +297,7 @@ void calc_mask(calcinfo_t &info, timinginfo_t &ti) {
 	uint8_t* out = (uint8_t*)info.ofinal.data;
 
 	// find class with maximum probability
-	if (strstr(info.modelname,"deeplab")) {
+	if (info.modeltype == ModelType::DeepLab) {
```

Comment:
What about using `switch`/`case` here? That way the compiler could actually check for completeness by testing if all cases are handled.

Reply:
yes!

Reply:
Also avoids the '=' when '==' was intended in a boolean expression bug :wink:

Reply:
As an aside: To avoid the `=` vs. `==` confusion you sometimes see the constant expression being written first. That way accidentally writing just one `=` gives you a compile error (e.g. `2 == foo` vs. `2 = foo`).